python/cli/visualization/visualizer_renderers/util.py

```diff
@@ -44,3 +44,12 @@ def getOrthographicProjectionMatrixOpenGL(left, right, bottom, top, near, far):
         [0, 0, 2 / (far - near), -(far + near) / (far - near)],
         [0, 0, 0, 1]
     ], dtype=np.float32)
+
+def getPerspectiveProjectionMatrixOpenGL(fovy, aspect, near, far):
+    f = 1.0 / np.tan(np.radians(fovy) / 2.0)
+    return np.array([
+        [f / aspect, 0, 0, 0],
+        [0, -f, 0, 0],
+        [0, 0, (far + near) / (far - near), -2.0 * far * near / (far - near)],
+        [0, 0, 1, 0]
+    ], dtype=np.float32)
```

Comment:
This projection matrix defnition has some sign differences w.r.t these sources:
https://gamedev.stackexchange.com/a/120355
https://www.songho.ca/opengl/gl_projectionmatrix.html

But I think this is because our camera coordinate system isn't the typical OpenGL one?

Reply:
The OpenGL "documentation" (=tutorials) and "conventions" (there actually aren't any) are a mess. The difference is indeed caused by how our camera coordinate convention (OpenCV) differs from some 3D programming folkore that has been repeatedly copy-pasted from tutorial to tutorial and now people think of as the OpenGL "convention".

If things look good with that projection matrix, and things like the depth test are not broken in some very visible way, let's go with that :+1: